benchmark/framework/agency.py

```diff
@@ -0,0 +1,191 @@
+
+
+class Agency:
+    """A testing wrapper to run and control an agent."""
+
+    def __init__(self, name: str = "my_eae", skills: List[dict] = None):
+        """Make an agency with optional name and skills."""
+        self.skills = skills or []
+        self.name = name
+
+        self.agent = self.make_agent(
+            self.name, [self.make_skill(**skill) for skill in self.skills]
+        )
+
+    def make_agent(
+        self, name: str = "my_eae", components: List[Component] = None
+    ) -> AEA:
+        """
+        Create AEA from name and already loaded components.
+
+        :return: AEA
+        """
+        components = components or []
+        builder = AEABuilder()
+
+        builder.set_name("my_aea")
+        builder.add_private_key(FETCHAI, "")
+
+        for component in components:
+            builder._resources.add_component(component)
+
+        aea = builder.build()
+        return aea
+
+    def make_skill(
+        self,
+        config: SkillConfig = None,
+        context: SkillContext = None,
+        handlers: Optional[Dict[str, Type[Handler]]] = None,
+    ) -> Skill:
+        """
+        Make a skill from optional config, context, handlers dict.
+
+        :return: Skill
+        """
+        handlers = handlers or {}
+        context = context or SkillContext()
+        config = config or SkillConfig()
+
+        handlers_instsances = {
+            name: cls(name=name, skill_context=context)
+            for name, cls in handlers.items()
+        }
+
+        skill = Skill(
+            configuration=config, skill_context=context, handlers=handlers_instsances
+        )
+
+        return skill
+
+    def dummy_message(
+        self,
+        dialogue_reference: Tuple[str, str] = ("", ""),
+        message_id: int = 1,
+        target: int = 0,
+        performative=DefaultMessage.Performative.BYTES,
+        content: Union[str, bytes] = "hello world!",
+    ) -> Message:
+        """
+        Construct simple message, all arguments are optional.
+
+        :return: Message
+        """
+        if isinstance(content, str):
+            content = content.encode("utf-8")
+
+        return DefaultMessage(
+            dialogue_reference=dialogue_reference,
+            message_id=message_id,
+            target=target,
+            performative=performative,
+            content=content,
+        )
+
+    def dummy_envelope(
+        self,
+        to: str = "test",
+        sender: str = "test",
+        protocol_id: ProtocolId = DefaultMessage.protocol_id,
+        message: Message = None,
+    ) -> Envelope:
+        """
+        Create envelope, if message is not passed use .dummy_message method.
+
+        :return: Envelope
+        """
+        message = message or self.dummy_message()
+        return Envelope(
+            to=to,
+            sender=sender,
+            protocol_id=protocol_id,
+            message=DefaultSerializer().encode(message),
```

Comment:
Since this is hardcoded as the default protocol here, you should not expose the protocol_id as an argument to the method as this can lead to a mismatch in serializer and protocol id

Reply:
In the same argument, there needs to be an assertion that the message is a `DefaultMessage` and the func signature needs updating.

Reply:
agree, but at the moment the main idea is to test framework, not every message type. I suppose tests will use only one dummy envelope all the time.